src/gui/guiChatConsole.cpp

```diff
@@ -328,6 +349,14 @@ void GUIChatConsole::drawText()
 			core::rect<s32> destrect(
 				x, y, x + m_fontsize.X * fragment.text.size(), y + m_fontsize.Y);
 
+
+			core::recti rect;
+			if (m_scrollbar->isVisible())
+				rect = core::rect<s32> (0, 0, m_screensize.X - getScrollbarSize(Environment), m_height);
+			else
+				rect = AbsoluteClippingRect;
```

Comment:
Please move this code outside the two `for` loops. No need to reevaluate them every time.